Kernel/CommandLine.cpp

```diff
@@ -59,14 +59,19 @@ UNMAP_AFTER_INIT void CommandLine::add_arguments(Vector<StringView> const& args)
             continue;
         }
 
-        auto pair = str.split_view('=');
-        VERIFY(pair.size() == 2 || pair.size() == 1);
-
-        if (pair.size() == 1) {
-            m_params.set(pair[0], ""sv);
-        } else {
-            m_params.set(pair[0], pair[1]);
-        }
+        // Note: The kernel command line might include entries like
+        // ```
+        // video=HDMI-A-1:1920x1080M@30D,margin_left=48,margin_right=48,margin_top=48,margin_bottom=48
```

Comment:
are you sure itʼs `video` value with csv inside not separate csv at large?

it seems for me that you (as well might) have here a command that you should split with a comma first and then each of the key-pair values need to be split for equality sign… (thatʼs at least that i see here, in almost all other cases).

Reply:
The kernel command line is a space separated list of keys and optional values,
This comment just shows part of the full command line, the part that would have broken with the previous implementation

I imagine that this is just a long list if attributes the connected display has on the raspberry pi

Reply:
i think i found it:
— https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/configuration/kernel-command-line-config.adoc
also this one (end of the first section):
— https://docs.kernel.org/fb/modedb.html

somewhat ill-defined for me but thatʼs exactly the case you described. you might add those for reference here (one or both, or something more/else?).

i guess thatʼs should be a good to go for me (until something funny pops up! 😅).

Reply:
Whoever parses that value can make it their problem to handle the composite string.

Reply:
yup! the thing was that if you see comma-separated key-values… well, thatʼs not the case here in the end. 😁